app/src/main/java/com/fish/monsters/features/history/ui/components/HistoryItem.kt

```diff
@@ -0,0 +1,164 @@
+        indentationSize = DpSize(12.dp, 31.dp),
+        contentPadding = PaddingValues(13.dp, 14.dp),
+    ) {
+        Row(Modifier.fillMaxSize()) {
+            Column(Modifier.width(200.dp)) {
+                Row(
+                    verticalAlignment = Alignment.CenterVertically
+                ) {
+                    Icon(
+                        imageVector = Icons.Default.CalendarMonth,
+                        contentDescription = "Calendar Icon",
+                        tint = DarkPrimaryColor,
+                        modifier = Modifier.width(16.dp)
+                    )
+                    Text(text = contestInfo.date, Modifier.padding(3.dp, 0.dp))
+                }
+                Spacer(
+                    modifier = Modifier
+                        .fillMaxHeight()
+                        .weight(1f)
+                )
+                Row {
+                    val difficultyStringRes = when (contestInfo.difficultyLevel) {
+                        DifficultyLevel.LOW -> R.string.low_difficulty
+                        DifficultyLevel.MEDIUM -> R.string.medium_difficulty
+                        DifficultyLevel.HIGH -> R.string.high_difficulty
+                    }
+                    Text(
+                        text = stringResource(id = difficultyStringRes),
+                        color = contestInfo.difficultyLevel.color
+                    )
+                }
+            }
+            Column(Modifier.width(200.dp), horizontalAlignment = Alignment.End) {
+                Row(
+                    verticalAlignment = Alignment.CenterVertically
+                ) {
+                    Icon(
+                        imageVector = Icons.Default.SportsScore,
+                        contentDescription = "Score Icon",
+                        tint = DarkPrimaryColor,
+                        modifier = Modifier.width(18.dp)
+                    )
+                    Spacer(Modifier.width(4.dp))
+                    Text(text = contestInfo.points.toString())
+                }
+                Spacer(
+                    modifier = Modifier
+                        .fillMaxHeight()
+                        .weight(1f)
+                )
+                Row(
+                    verticalAlignment = Alignment.CenterVertically
+                ) {
+                    Icon(
+                        imageVector = Icons.Default.AccessTime,
+                        contentDescription = "Time Icon",
+                        tint = DarkPrimaryColor,
+                        modifier = Modifier.width(14.dp)
+                    )
+                    Spacer(Modifier.width(4.dp))
+                    Text(
+                        text = buildAnnotatedString {
+                            if (contestInfo.duration.hours > 0) {
+                                append("${contestInfo.duration.hours}h ")
+                            }
+                            append("${contestInfo.duration.minutes}min")
```

Comment:
zróbmy nową funkcje `toString()` w `Duration` dla tego działania, i operujmy tam na Stringach zwychłych `AnnotatedString` służy do stylowania kawałków textu wewnątrz textu, tutaj zmieniamy sam w sobie text, możemy to wyciągnąć na luzie